### PR TITLE
Add option to disable zsh global compinit

### LINKED ISSUE #855: Option to disable zsh global compinit
If a user wants to add to `fpath` or call `compinit` with other options (e.g. a custom dump filename), they should be able to. NixOS's `programs.zsh.enableGlobalCompInit` accomplishes this.

### PR DESCRIPTION
**Pull request by mrdomino**
Fixes #855.

### CODE REVIEW

File: modules/programs/zsh/default.nix [hunk 1]
```diff
@@ -70,6 +70,18 @@ in
       description = lib.mdDoc "Enable bash completion for all interactive zsh shells.";
     };
 
+    programs.zsh.enableGlobalCompInit = mkOption {
+      type = types.bool;
+      default = cfg.enableCompletion;
```
Comment (Enzime) [inline modules/programs/zsh/default.nix:75]:
```suggestion
      default = cfg.enableCompletion;
      defaultText = literalExpression "config.${opt.enableCompletion}";
```

**Review by Enzime — CHANGES_REQUESTED** [review]

**Comment by mrdomino** [comment]
failure looks spurious

**Comment by Enzime** [comment]
Can you squash the two commits together?

**Comment by mrdomino** [comment]
Done.

**Review by Enzime — APPROVED** [review]
LGTM

**Pull request merged**